core/variable_reduction_operations.cpp

```diff
@@ -75,7 +75,7 @@ void flatten_impl(const VariableProxy &summed, const VariableConstProxy &var,
       summed, var, mask,
       overloaded{
           [](auto &a, const auto &b, const auto &mask_) {
-            if (!mask_)
+            if (mask_)
```

Comment:
was this a bug we had here?

Reply:
No I just inverted the logic to match that of `sum_impl` etc.